crates/sequencer/src/mempool.rs

```diff
@@ -85,6 +85,9 @@ impl<C: sov_modules_api::Context> CitreaMempool<C> {
         let validator = TransactionValidationTaskExecutor::eth_builder(Arc::new(chain_spec))
             .no_cancun()
             .no_eip4844()
+            // TODO: if we ever increase block gas limits, we need to pull this from
+            // somewhere else
+            .set_block_gas_limit(evm_config.block_gas_limit)
```

Comment:
bc we forgot to add this, the mempool thought our block gas limit is 30m, then accepted txs with over 8m gas.

but they can never be included in  a block